### ISSUE TITLE
[tune] Implement population based training as a ray.tune TrialScheduler

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#1289]
We should implement https://deepmind.com/blog/population-based-training-neural-networks/ as a custom TrialScheduler in ray.tune.

PBT implements hyperparameter perturbations within individual algorithm runs, which can substantially improve efficiency compared to current schedulers that fix parameters starting from the beginning of the run (e.g. MedianStoppingRule, HyperBand). In addition, this automatically discovers reasonable hyperparam schedules, which is something completely not possible with current schedulers.

The implementation could look something like this:
- We use the ray.tune json spec as the initial model population
- The trial scheduler periodically perturbs promising trials / removes poor trials as described in the paper: https://arxiv.org/abs/1711.09846

This would be the first trial scheduler that creates new trials during runs, so there might be some kinks to work out with how results are logged. In addition, we might need to provide some way to annotate params that should be perturbed.

**Comment by Kaahan** [comment]
Looking into this right now 👍
